### DISCUSSION TITLE
Locking content

### DISCUSSION THREAD

**Discussion by jameswilliamknight** [AppFlowy-IO/AppFlowy#2982] [Ideas]
Hi there.

I haven't done much research beyond searching the docs for the keyword "Lock", so I thought I'd start this conversation and at least docs could get a bump, if not start discussion on a feature like this.

One of my pet peeves with notion was always how it felt like I had either no control over the interaction with a page (locking it), or everything was unlocked and it felt prone to accidents;

- moving content
- moving content to another page and not knowing where
- providing less technically literate friends/family access, but in order to make a page function, you'd be giving them edit access.

What I'd like to see in either Notion or appflowy, is a locking system that's fine grained enough to act on the component level. To edit the lock states, you would enter a pseudo-edit mode (lock-edit mode, if you will), where other kinds of edits are disabled temporarily.

It was an idea floating around, just interested in thoughts from others. Thanks

**Comment by annieappflowy** [comment]
Could you please elaborate on the component level you suggested? Maybe provide me with a few examples.
And how do you envision a locked component to be unlocked?

**Reply by EclipseNow** [reply]
Hi all, I’m hoping Appflowy can evolved to replace Onenote for my security needs. I need all my bank and medical and government admin on me at all times. For security to compete with Onenote my dream app needs:-
* password locking - and automatic locking after 5 minutes AFK. I review my notes a lot when out and about - and if a guy distracts me while his thieving mate grabs my open phone from the Diner table with my notes app open - I’m done! If it has AFK, I might have a chance that it’s locked by the time they run away and have a look.
* store the notes in an encrypted database on my hard-drive as well as the cloud. EG: Obsidian has a password plugin - but if someone gets my computer open they could look up the Obsidian notes file - and there they are in markdown!
* Able to share certain “Notebooks” with a visitor requiring their own password for those sections - but not having access to the whole thing. EG: Trilium got so close - but it's only single user. If I share that password - it unlocks ALL NOTES to the visitor.
* This last one is optional - but I’d love there to be a Freemium model with cloud syncing as well where the first 5 to 15 Gigs cloud syncing are free. (Think Microsoft & Google). 
I’m not alone - there are many seeking passwords & AFK & secure sharing. We’ve got used to it. It’s in Onenote - after all!
Keep me posted! 
Cheers. :-)
